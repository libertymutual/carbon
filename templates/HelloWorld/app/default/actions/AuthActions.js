import * as ActionTypes from '../actions/ActionTypes';
import { NativeModules } from 'react-native';

const { Auth } = NativeModules;

export type AuthActionsType = {
  login: () => ((Object) => void, () => Object) => {},
logout: () => () => { },
};


/**
 * Auth related action creators
 */
export default {

  /**
   * Logs the user in
   * @returns {function} A function which dispatches a LOGIN action
   */
  login() {
    return (dispatch: Object => void, getState: () => Object) => {
      const username = getState().username;
      const password = getState().password;
      Auth.login(username, password).then((isLoggedIn) => {
        if (isLoggedIn) {
          dispatch({ type: ActionTypes.LOGIN })
        }
      })
    }
  },

  /**
   * Logs the user out
   * @returns {function} A function which dispatches a LOGOUT action
   */
  logout() {
    return function (dispatch) {
      dispatch({ type: ActionTypes.LOGOUT });
    };
  },

  updateAuthUrl(authUrl) {
    return function (dispatch) {
      dispatch({ type: ActionTypes.AUTH_UPDATE_AUTH_URL, authUrl })
    }
  },

  updateClientId(clientId) {
    return function (dispatch) {
      dispatch({ type: ActionTypes.AUTH_UPDATE_CLIENT_ID, clientId })
    }
  },

  updateClientSecret(clientSecret) {
    return function (dispatch) {
      dispatch({ type: ActionTypes.AUTH_UPDATE_CLIENT_SECRET, clientSecret })
    }
  },

  updateConfigName(configName) {
    return function (dispatch) {
      dispatch({ type: ActionTypes.AUTH_UPDATE_CONFIG_NAME, configName })
    }
  },

  updatePassword(password) {
    return function (dispatch) {
      dispatch({ type: ActionTypes.AUTH_UPDATE_PASSWORD, password })
    }
  },

  updateUsername(username) {
    return function (dispatch) {
      dispatch({ type: ActionTypes.AUTH_UPDATE_USERNAME, username })
    }
  },

  updateValidatorId(validatorId) {
    return function (dispatch) {
      dispatch({ type: ActionTypes.AUTH_UPDATE_VALIDATOR_ID, validatorId })
    }
  },
};
