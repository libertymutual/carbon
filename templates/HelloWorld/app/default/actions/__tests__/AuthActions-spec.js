import 'react-native';
import React from 'react';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import AuthActions from '../AuthActions';

describe('AuthActions tests', () => {

  const middlewares = [thunk];
  const mockStore = configureMockStore(middlewares);
  let store;
  let dispatch;

  beforeEach(() => {
    // Sets up a new store
    store = mockStore();
    dispatch = store.dispatch;
  });

  test('login returns a function', () => {
    var dispatcher = AuthActions.login();
    expect(typeof dispatcher).toMatchSnapshot();
  });

  test('login resturs a LOGIN action', () => {
    var dispatcher = AuthActions.login()
    expect(store.getActions()).toMatchSnapshot();
  });

  test('logout returns a function', () => {
    var dispatcher = AuthActions.logout();
    expect(typeof dispatcher).toMatchSnapshot();
  });

  test('logout function dispatches a LOGOUT action', () => {
    var dispatcher = AuthActions.logout();
    dispatch(dispatcher);
    expect(store.getActions()).toMatchSnapshot();
  });

  test('updateAuthUrl returns a function', () => {
    var dispatcher = AuthActions.updateAuthUrl();
    expect(typeof dispatcher).toMatchSnapshot();
  });

  test('updateAuthUrl dispatches a AUTH_UPDATE_AUTH_URL action', () => {
    var dispatcher = AuthActions.updateAuthUrl();
    dispatch(dispatcher);
    expect(store.getActions()).toMatchSnapshot();
  });

  test('updateClientId returns a function', () => {
    var dispatcher = AuthActions.updateClientId();
    expect(typeof dispatcher).toMatchSnapshot();
  });

  test('updateClientId function dispatches a AUTH_UPDATE_CLIENT_ID action', () => {
    var dispatcher = AuthActions.updateClientId();
    dispatch(dispatcher);
    expect(store.getActions()).toMatchSnapshot();
  });

  test('updateClientSecret returns a function', () => {
    var dispatcher = AuthActions.updateClientSecret();
    expect(typeof dispatcher).toMatchSnapshot();
  });

  test('updateClientSecret function dispatches a AUTH_UPDATE_CLIENT_SECRET action', () => {
    var dispatcher = AuthActions.updateClientId();
    dispatch(dispatcher);
    expect(store.getActions()).toMatchSnapshot();
  });

  test('updateConfigName returns a function', () => {
    var dispatcher = AuthActions.updateConfigName();
    expect(typeof dispatcher).toMatchSnapshot();
  });

  test('updateConfigName function dispatches a AUTH_UPDATE_CONFIG_NAME action', () => {
    var dispatcher = AuthActions.updateClientId();
    dispatch(dispatcher);
    expect(store.getActions()).toMatchSnapshot();
  });

  test('updatePassword returns a function', () => {
    var dispatcher = AuthActions.updatePassword();
    expect(typeof dispatcher).toMatchSnapshot();
  });

  test('updatePassword function dispatches a AUTH_UPDATE_PASSWORD action', () => {
    var dispatcher = AuthActions.updatePassword();
    dispatch(dispatcher);
    expect(store.getActions()).toMatchSnapshot();
  });

  test('updateUsername returns a function', () => {
    var dispatcher = AuthActions.updateUsername();
    expect(typeof dispatcher).toMatchSnapshot();
  });

  test('updateUsername function dispatches a AUTH_UPDATE_USERNAME action', () => {
    var dispatcher = AuthActions.updateUsername();
    dispatch(dispatcher);
    expect(store.getActions()).toMatchSnapshot();
  });

  test('updateValidatorId returns a function', () => {
    var dispatcher = AuthActions.updateValidatorId();
    expect(typeof dispatcher).toMatchSnapshot();
  });

  test('updateValidatorId function dispatches a AUTH_UPDATE_VALIDATOR_ID action', () => {
    var dispatcher = AuthActions.updateValidatorId();
    dispatch(dispatcher);
    expect(store.getActions()).toMatchSnapshot();
  });
});
