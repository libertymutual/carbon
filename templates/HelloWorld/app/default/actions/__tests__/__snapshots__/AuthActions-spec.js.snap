// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthActions tests login resturs a LOGIN action 1`] = `Array []`;

exports[`AuthActions tests login returns a function 1`] = `"function"`;

exports[`AuthActions tests logout function dispatches a LOGOUT action 1`] = `
Array [
  Object {
    "type": "logout",
  },
]
`;

exports[`AuthActions tests logout returns a function 1`] = `"function"`;

exports[`AuthActions tests updateAuthUrl dispatches a AUTH_UPDATE_AUTH_URL action 1`] = `
Array [
  Object {
    "authUrl": undefined,
    "type": "auth_update_auth_url",
  },
]
`;

exports[`AuthActions tests updateAuthUrl returns a function 1`] = `"function"`;

exports[`AuthActions tests updateClientId function dispatches a AUTH_UPDATE_CLIENT_ID action 1`] = `
Array [
  Object {
    "clientId": undefined,
    "type": "auth_update_client_id",
  },
]
`;

exports[`AuthActions tests updateClientId returns a function 1`] = `"function"`;

exports[`AuthActions tests updateClientSecret function dispatches a AUTH_UPDATE_CLIENT_SECRET action 1`] = `
Array [
  Object {
    "clientId": undefined,
    "type": "auth_update_client_id",
  },
]
`;

exports[`AuthActions tests updateClientSecret returns a function 1`] = `"function"`;

exports[`AuthActions tests updateConfigName function dispatches a AUTH_UPDATE_CONFIG_NAME action 1`] = `
Array [
  Object {
    "clientId": undefined,
    "type": "auth_update_client_id",
  },
]
`;

exports[`AuthActions tests updateConfigName returns a function 1`] = `"function"`;

exports[`AuthActions tests updatePassword function dispatches a AUTH_UPDATE_PASSWORD action 1`] = `
Array [
  Object {
    "password": undefined,
    "type": "auth_update_password",
  },
]
`;

exports[`AuthActions tests updatePassword returns a function 1`] = `"function"`;

exports[`AuthActions tests updateUsername function dispatches a AUTH_UPDATE_USERNAME action 1`] = `
Array [
  Object {
    "type": "auth_update_username",
    "username": undefined,
  },
]
`;

exports[`AuthActions tests updateUsername returns a function 1`] = `"function"`;

exports[`AuthActions tests updateValidatorId function dispatches a AUTH_UPDATE_VALIDATOR_ID action 1`] = `
Array [
  Object {
    "type": "auth_update_validator_id",
    "validatorId": undefined,
  },
]
`;

exports[`AuthActions tests updateValidatorId returns a function 1`] = `"function"`;
